connect/tunnel_api/tunnel_api.go

```diff
@@ -0,0 +1,853 @@
+		Port: strconv.Itoa(int(tunnel.Options.ServerPort)),
+	}
+
+	err = openvpnConfTemplate.Execute(&openvpnConf, configdata)
+	if err != nil {
+		return err
+	}
+
+	configmap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: *&tunnel.Options.Namespace,
+		},
+		Data: map[string]string{
+			"openvpn.conf": openvpnConf.String(),
+		},
+	}
+
+	openvpnService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: *&tunnel.Options.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       serviceName,
+					Protocol:   corev1.ProtocolTCP,
+					Port:       tunnel.Options.ServerPort,
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: tunnel.Options.ServerPort},
+				},
+			},
+			Selector: deploymentLabels,
+			Type:     corev1.ServiceTypeLoadBalancer,
+		},
+	}
+
+	proxiedService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "proxied-cluster",
+			Namespace: *&tunnel.Options.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "proxy",
+					Protocol:   corev1.ProtocolTCP,
+					Port:       int32(8443),
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 8443},
+				},
+			},
+			Selector: deploymentLabels,
+			Type:     corev1.ServiceTypeClusterIP,
+		},
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: *&tunnel.Options.Namespace,
+		},
+		Data: map[string][]byte{
+			"ca.crt":     tunnel.Options.CACrt.Bytes(),
+			"dh.pem":     tunnel.Options.RSADHKey.Bytes(),
+			"server.crt": tunnel.Options.ServerCrt.Bytes(),
+			"server.key": tunnel.Options.ServerKey.Bytes(),
+		},
+	}
+
+	mode := int32(0400)
+	volumes := []v1.Volume{
+		{
+			Name: serviceName,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					DefaultMode: &mode,
+					SecretName:  serviceName,
+				},
+			},
+		},
+		{
+			Name: serviceConfig,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: serviceName,
+					},
+				},
+			},
+		},
+	}
+
+	volumeMounts := []v1.VolumeMount{
+		{
+			Name:      serviceName,
+			MountPath: "/certs",
+		},
+		{
+			Name:      serviceConfig,
+			MountPath: "/openvpn.conf",
+			SubPath:   "openvpn.conf",
+		},
+	}
+
+	trueBool := true
+	containers := []corev1.Container{
+		{
+			Name:  serviceName,
+			Image: *&tunnel.Options.ServerImage,
+			Command: []string{
+				"bash",
+				"-c",
+				"mkdir /tmp/openvpn && openvpn --config /openvpn.conf",
+			},
+			VolumeMounts:    volumeMounts,
+			SecurityContext: &corev1.SecurityContext{Privileged: &trueBool},
+		},
+		{
+			Name:  "socat",
+			Image: *&tunnel.Options.ServerImage,
+			Command: []string{
+				"bash",
+				"-c",
+				"socat TCP-LISTEN:8443,fork,reuseaddr TCP:192.168.123.6:8443",
```

Comment:
A network to link the two containers. I thought about making the network range configurable, but in all my testing, with multiple connections, etc. I saw no need to modify it. I did use the env vars on the client container for the API endpoint event though it's usually 172.30.0.1 in our testing, because it is configurable. It might not be often that is changed, but it could happen.

Reply:
I see, that makes sense wonder if just a follow-up issue makes sense that folks can +1 if it becomes something they need?

Reply:
https://github.com/konveyor/crane-lib/issues/88